impl/src/main/java/org/pentaho/platform/genericfile/providers/repository/RepositoryFileProvider.java

```diff
@@ -492,6 +492,15 @@ public List<IGenericFile> getDeletedFiles() {
       .collect( Collectors.toList() );
   }
 
+  @Override
+  public void permanentDeleteFiles( @NonNull List<String> fileIds ) throws OperationFailedException {
+    try {
```

Comment:
A special parsing of the received file **path** needs to be done, so as to extract the repository file id from the deleted repository path. If such an id cannot be recovered then the specified file is not deleted and an error be thrown.